pkg/nsx/services/vpc/vpc.go

```diff
@@ -389,22 +389,48 @@ func (s *VPCService) GetAVISubnetInfo(vpc model.Vpc) (string, string, error) {
 	return path, cidr, nil
 }
 
-func (s *VPCService) GetNSXLBSNATIP(vpc model.Vpc) (string, error) {
+func (s *VPCService) GetNSXLBSNATIPForCTGW(vpc model.Vpc) (string, error) {
 	log.V(2).Info("Getting VPC NSX LB SNAT IP", "VPC", *vpc.Id)
 	_, err := common.ParseVPCResourcePath(*vpc.Path)
 	if err != nil {
 		return "", err
 	}
 
 	realizeService := realizestate.InitializeRealizeState(s.Service)
-	tier1UpLinkIP, err := realizeService.GetPolicyTier1UplinkPortIP(*vpc.Path)
+	parts := strings.Split(*vpc.Path, "/vpcs/")
+	if len(parts) != 2 {
+		return "", fmt.Errorf("invalid VPC path format: %s", *vpc.Path)
+	}
+	realizedPath := fmt.Sprintf("%s/realized-state/enforcement-points/default/vpcs/%s/gateway-interface", parts[0], parts[1])
+	tier1UpLinkIP, err := realizeService.GetPolicyServiceInterfaceSNATIP(realizedPath)
 	if err != nil {
 		log.Error(err, "Failed to get VPC NSX LB SNAT IP", "VPC", *vpc.Id)
 		return "", err
 	}
 	return tier1UpLinkIP, nil
 }
 
+func (s *VPCService) GetNSXLBSNATIPForDTGW(vpc model.Vpc) (string, error) {
```

Comment:
GetNSXLBSNATIPForDTGW logic basically is the same as  GetNSXLBSNATIPForCTGW, in my opinion, it's not necessary to make two functions and the API call in two functions are the same as well.

Reply:
fixed